.github/workflows/doc.yml

```diff
@@ -0,0 +1,32 @@
+name: doc
+
+on:
+  pull_request:
+    branches: ["master"]
+
+permissions:
+  contents: read
+
+env:
```

Comment:
Is there specific reasons to use 3.11?
If not, let's use 3.12 as it seems to be the fastest.